src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -0,0 +1,542 @@
+                }
+                walkTo = pos;
+            }
+
+            if (walkToBlock(walkTo))
+            {
+                return getState();
+            }
+
+            final boolean transfered = tryTransferFromPosToCook(walkTo, TileEntityFurnace::isItemFuel);
+            if (!transfered)
+            {
+                walkTo = null;
+                return START_WORKING;
+            }
+            walkTo = null;
+        }
+
+        return COOK_COOK_FOOD;
+    }
+
+    private AIState retrieve()
+    {
+        if (walkTo == null)
+        {
+            return START_WORKING;
+        }
+
+        if (walkToBlock(walkTo))
+        {
+            return getState();
+        }
+
+        final TileEntity entity = world.getTileEntity(walkTo);
+        if (!(entity instanceof TileEntityFurnace)
+                || ((TileEntityFurnace) entity).isBurning()
+                || (ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(2))
+                && ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(0))))
+        {
+            walkTo = null;
+            return START_WORKING;
+        }
+        walkTo = null;
+
+        InventoryUtils.transferItemStackIntoNextFreeSlotInItemHandlers(
+                new InvWrapper((TileEntityFurnace) entity), 2,
+                new InvWrapper(worker.getInventoryCitizen()));
+
+        if (!ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(0)))
+        {
+            if (!InventoryUtils.hasItemInItemHandler(new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel))
+            {
+                walkTo = null;
+                return COOK_GET_FIREWOOD;
+            }
+
+            InventoryUtils.transferItemStackIntoNextFreeSlotInItemHandlers(
+                    new InvWrapper(worker.getInventoryCitizen()),
+                    InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel),
+                    new InvWrapper((TileEntityFurnace) entity));
+        }
+
+        incrementActionsDone();
+        setDelay(STANDARD_DELAY);
+        return START_WORKING;
+    }
+
+    private AIState gatherFoodFromBuilding()
+    {
+        if(needsCurrently == null)
+        {
+            needsCurrently = isFood;
+        }
+
+        if(InventoryUtils.hasItemInItemHandler(new InvWrapper(worker.getInventoryCitizen()), needsCurrently))
+        {
+            return COOK_SERVE;
+        }
+
+        final BlockPos pos = getOwnBuilding().getTileEntity().getPositionOfChestWithItemStack(needsCurrently);
+        if (pos == null)
+        {
+            return START_WORKING;
+        }
+
+        if (walkToBlock(pos))
+        {
+            return getState();
+        }
+
+        final boolean transfered = tryTransferFromPosToCook(pos, needsCurrently);
+        if (transfered)
+        {
+            return COOK_SERVE;
+        }
+        //todo
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

Reply:
?